assets/js/googlesitekit/api/middleware/preloading.test.js

```diff
@@ -0,0 +1,163 @@
+		const firstRequestURI = 'test/path/a';
+		const secondRequestURI = 'test/path/b';
+
+		const preloadingMiddleware = createPreloadingMiddleware(
+			preloadedData
+		);
+
+		const firstResponseNext = jest.fn();
+		const secondResponseNext = jest.fn();
+
+		const firstResponse = await preloadingMiddleware( { method: 'GET', path: firstRequestURI }, firstResponseNext );
+		expect( firstResponse ).toEqual( preloadedData[ firstRequestURI ].body );
+		expect( firstResponseNext ).not.toHaveBeenCalled();
+
+		const firstResponseAgain = await preloadingMiddleware( { method: 'GET', path: firstRequestURI }, firstResponseNext );
+		expect( firstResponseAgain ).toBeUndefined();
+		expect( firstResponseNext ).toHaveBeenCalled();
+
+		const secondResponse = await preloadingMiddleware( { method: 'GET', path: secondRequestURI }, secondResponseNext );
+		expect( secondResponse ).toEqual( preloadedData[ secondRequestURI ].body );
+		expect( secondResponseNext ).not.toHaveBeenCalled();
+
+		const secondResponseAgain = await preloadingMiddleware( { method: 'GET', path: secondRequestURI }, secondResponseNext );
+		expect( secondResponseAgain ).toBeUndefined();
+		expect( secondResponseNext ).toHaveBeenCalled();
+	} );
+
+	it( 'does nothing and calls next middleware when no preloaded response exists for the request', async () => {
+		const requestURI = 'test/path/a';
+		const next = jest.fn();
+		const preloadingMiddleware = createPreloadingMiddleware( {} );
+		const requestOptions = {
+			method: 'GET',
+			path: requestURI,
+		};
+		await preloadingMiddleware( requestOptions, next );
+		expect( next ).toHaveBeenCalled();
+	} );
+
+	it( 'returns a preloaded response only once for each preloaded request', async () => {
+		const requestURI = 'test/path/a';
+		const preloadingMiddleware = createPreloadingMiddleware(
+			preloadedData
+		);
+
+		const requestOptions = {
+			method: 'GET',
+			path: requestURI,
+		};
+		const next = jest.fn();
+		const firstResponse = await preloadingMiddleware( requestOptions, next );
+		expect( firstResponse ).toEqual( preloadedData[ requestURI ].body );
+		expect( next ).not.toHaveBeenCalled();
+
+		const secondRequest = await preloadingMiddleware( requestOptions, next );
+		expect( secondRequest ).toBeUndefined();
+		expect( next ).toHaveBeenCalled();
+	} );
+
+	it( 'calls the next middleware', async () => {
+		const firstRequestURI = 'test/path/a';
+		const secondRequestURI = 'test/path/b';
+		const preloadingMiddleware = createPreloadingMiddleware(
+			preloadedData,
+			10
+		);
+		const requestOptions = {
+			method: 'GET',
+			path: firstRequestURI,
+		};
+		const next = jest.fn();
+
+		const firstResponse = await preloadingMiddleware( requestOptions, next );
+		expect( firstResponse ).toEqual( preloadedData[ firstRequestURI ].body );
+		expect( next ).not.toHaveBeenCalled();
+
+		jest.runAllTimers();
+		// Confirm that responses after the timeout don't run hit the middleware.
```

Comment:
There's a bit of a typo in this sentence ("run hit"); how about this?
```suggestion
		// Confirm that invocations after the timeout ignore preloaded response cache.
```